l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -305,9 +316,11 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
             _assetData: _assetData
         });
 
-        bytes32 txDataHash = keccak256(abi.encode(_prevMsgSender, assetInfo, _assetData));
+        (uint256 _depositAmount, ) = abi.decode(_assetData, (uint256, address));
```

Comment:
No, this is bad. ``_assetData`` might not be encoded like that, this only works for legacy deposits.

Reply:
Updated